### ISSUE TITLE
file() function failed to follow the URL redirection

### ISSUE THREAD

**Issue by DongzeHE** [nextflow-io/nextflow#4131]
## Bug report 

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior

(file() function supposed to download the file when providing a remote URL but did not)

### Steps to reproduce the problem

(file(http://mitra.stanford.edu/kundaje/akundaje/release/blacklists/mm10-mouse/mm10.blacklist.bed.gz))

### Program output 

(
The downloaded file is like 
```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="https://mitra.stanford.edu/kundaje/akundaje/release/blacklists/mm10-mouse/mm10.blacklist.bed.gz">here</a>.</p>
<hr>
<address>Apache/2.4.29 (Ubuntu) Server at mitra.stanford.edu Port 80</address>
</body></html>
```
)

### Environment 

* Nextflow version: =[22.10.4] 
* Java version: [17.0.4]
* Operating system: [Linux]
* Bash version: (4.2.46)

### Additional context

(Add any other context about the problem here)

**Comment by pditommaso** [comment]
May be related to https://github.com/nextflow-io/nextflow/issues/4092

**Comment by pditommaso** [comment]
Thanks for reporting this issue. Solved by f59a60d6
